src/assets/src/components/addQueue.tsx

```diff
@@ -0,0 +1,388 @@
+                    <Tab.Content>
+                        <Tab.Pane eventKey='general'>
+                            <GeneralTab
+                                disabled={props.disabled}
+                                name={props.name}
+                                nameValidationResult={props.nameValidationResult}
+                                onChangeName={props.onChangeName}
+                                description={props.description}
+                                descriptValidationResult={props.descriptValidationResult}
+                                onChangeDescription={props.onChangeDescription}
+                                backends={props.backends}
+                                allowedMeetingTypes={props.allowedMeetingTypes}
+                                allowedIsInvalid={props.allowedIsInvalid}
+                                onChangeAllowed={props.onChangeAllowed}
+                                onNext={props.onGeneralNextClick}
+                            />
+                        </Tab.Pane>
+                        <Tab.Pane eventKey='hosts'>
+                            <ManageHostsTab
+                                disabled={props.disabled}
+                                currentUser={props.currentUser}
+                                hosts={props.hosts}
+                                onNewHost={props.onNewHost}
+                                checkHostError={props.checkHostError}
+                                onChangeHosts={props.onChangeHosts}
+                                onBack={props.onBackClick}
+                                onFinish={props.onFinishClick}
+                            />
+                        </Tab.Pane>
+                    </Tab.Content>
+                </Col>
+            </Row>
+        </Tab.Container>
+    );
+}
+
+export function AddQueuePage(props: PageProps) {
+    if (!props.user) {
+        redirectToLogin(props.loginUrl);
+    }
+
+    // Set up page state
+    const [activeKey, setActiveKey] = useState('general');
+
+    const [name, setName] = useState('');
+    const [nameValidationResult, setNameValidationResult] = useState(undefined as undefined | ValidationResult);
+    const [description, setDescription] = useState('');
+    const [descriptValidationResult, setDescriptValidationResult] = useState(undefined as undefined | ValidationResult);
+    const [allowedMeetingTypes, setAllowedMeetingTypes] = useState(new Set() as Set<string>);
+    const [allowedIsInvalid, setAllowedIsInvalid] = useState(undefined as undefined | boolean);
+
+    const [hosts, setHosts] = useState([props.user] as User[]);
+
+    // Set up API interactions
+    const checkHost = async (uniqname: string): Promise<User> => {
+        return await confirmUserExists(uniqname);
+    }
+    const [doCheckHost, checkHostLoading, checkHostError] = usePromise(
+        checkHost,
+        (value: User) => setHosts([...hosts].concat(value))
+    );
```

Comment:
@jlost, you'll notice that I've implemented this form so that we check for the existence of hosts when they are entered (using `doCheckHost`), but we subsequently actually create the hosts as part of the queue creation. I decided that it would simplify the implementation and make more sense to not create the queue, hosts as we go along, but this opens up the (tiny) possibility of a user being deleted after it was confirmed but before the queue was created. This results in an `Add Queue` error that's pretty obscure, referencing the pk of the nonexistent user. Do you think we need to/have any ideas on how might handle this case more elegantly?

Reply:
Cool, I generally agree that the exception is quite rare. If possible, it seems like it might be nice to improve the error message. Is there a way to control that in the backend? Looking into it. At least indicating that the entity that doesn't exist is a host would be good.

Reply:
@jlost, I'm going to create a new issue about this so we don't forget about it, but it can remain low priority or eventually end up as `wontfix`.

Reply:
See #172.